content/backend/typescript-apollo/9-deployment.md

```diff
@@ -0,0 +1,692 @@
+
+
+```bash(path=".../hackernews-typescript/")
+gh secret set HEROKU_EMAIL
+```
+
+</Instruction>
+
+
+
+> **Note:** If you make a mistake and need to update the value of a secret, running the `gh secret set` command again with the same secret name will update the value of that particular secret. 
+
+<Instruction>
+
+Similarly, create the `HEROKU_APP_NAME` secret using the name of the Heroku app you just created as the value. 
+
+```bash(path=".../hackernews-typescript/")
+gh secret set HEROKU_APP_NAME
+```
+
+</Instruction>
+
+
+You can check or verify the Heroku app name by running the following command in the root of your project: 
+
+```bash(path=".../hackernews-typescript/")
+heroku apps:info
+```
+
+Now you will need to retreive your `HEROKU_API_KEY`. This key is available under the **Account** tab of your Heroku [**Account Settings**](https://dashboard.heroku.com/account). Go to the [**Account Settings**](https://dashboard.heroku.com/account) and scroll down to **API Key**. Finally press the **Reveal** button to retreive the key. 
+
+![Heroku account API key](https://i.imgur.com/llG5CfB.png) 
+
+<Instruction>
+
+Set the `HEROKU_API_KEY` secret using the key retreived from your Heroku account. 
+
+```bash(path=".../hackernews-typescript/")
+gh secret set HEROKU_API_KEY
+```
+
+</Instruction>
+
+The final secret is `DATABASE_URL` which contains the connection string to your production database. This is already available in Heroku as an environment variable under the same name (`DATABASE_URL`). You can fetch it from Heroku using the following command:
+
+```bash(path=".../hackernews-typescript/")
+heroku config:get DATABASE_URL
+```
+
+<Instruction>
+
+Set the `DATABASE_URL` secret using the URL retreived from your Heroku app. 
+
+```bash(path=".../hackernews-typescript/")
+gh secret set DATABASE_URL
+```
+
+</Instruction>
+
+That's it! All the secrets have been set and it is now possible to deploy and execute your github action. If you want to see the name of the secrets available in your repository, you can do so using the `gh secret list` command. 
+
+
+<Instruction>
+
+Create a new commit and push all your changes to the remote repository on GitHub: 
+
+```bash(path=".../hackernews-typescript/")
+git add .
+git commit -m "create deployment github action workflow" 
+git push origin master
+```
+
+</Instruction>
+
+This should create and trigger the `deploy-hackernews-app-heroku` workflow on GitHub. You can trigger the workflow at any time creating an empty commit: 
+
+```bash(path=".../hackernews-typescript/")
+git commit --allow-empty -m "trigger build"
+git push
+```
+
+Once you have pushed a commit, quickly go to the **Actions** tab of your GitHub repository, and you should see that the workflow is _in progress_. 
+
+![Workflow in progress in the Actions tab](https://i.imgur.com/8xBtd3q.png)
```

Comment:
I think what you suggested is ideal. However, all the existing tutorials are using Imgur, so it would be somewhat time-consuming to refactor. My feeling is that we don't have the time to do this at the moment, but I'll bring this up with Niko.

Reply:
We wouldn't have to move **all** the assets but the assets for this tutorial. Most of the assets are hosted in the repository for the Prisma Blog. A possible option would be to host the images on GH – uploading them in an issue, discussion or PR and using the link they give you in the response 😅